spec/ddtrace/diagnostics/environment_logger_spec.rb

```diff
@@ -0,0 +1,268 @@
+
+        it { is_expected.to include tags: 'k1:v1,k2:v2' }
+      end
+
+      context 'with service configured' do
+        before { Datadog.configure { |c| c.service = 'svc' } }
+
+        it { is_expected.to include service: 'svc' }
+      end
+
+      context 'with version configured' do
+        before { Datadog.configure { |c| c.version = '1.2' } }
+
+        it { is_expected.to include dd_version: '1.2' }
+      end
+
+      context 'with debug enabled' do
+        before { Datadog.configure { |c| c.diagnostics.debug = true } }
+
+        it { is_expected.to include debug: true }
+      end
+
+      context 'with analytics enabled' do
+        before { Datadog.configure { |c| c.analytics_enabled = true } }
+
+        it { is_expected.to include analytics_enabled: true }
+      end
+
+      context 'with runtime metrics enabled' do
+        before { Datadog.configure { |c| c.runtime_metrics_enabled = true } }
+
+        it { is_expected.to include runtime_metrics_enabled: true }
+      end
+
+      context 'with partial flushing enabled' do
+        before { Datadog.configure { |c| c.tracer.partial_flush.enabled = true } }
+
+        it { is_expected.to include partial_flushing_enabled: true }
+      end
+
+      context 'with priority sampling enabled' do
+        before { Datadog.configure { |c| c.tracer.priority_sampling = true } }
+
+        it { is_expected.to include priority_sampling_enabled: true }
+      end
+
+      context 'with health metrics enabled' do
+        before { Datadog.configure { |c| c.diagnostics.health_metrics.enabled = true } }
+
+        it { is_expected.to include health_metrics_enabled: true }
+      end
+
+      context 'with agent connectivity issues' do
+        let(:response) { Datadog::Transport::InternalErrorResponse.new(ZeroDivisionError.new('msg')) }
+
+        it { is_expected.to include agent_error: include('ZeroDivisionError') }
+        it { is_expected.to include agent_error: include('msg') }
+      end
+
+      context 'with unix socket transport' do
+        before do
+          Datadog.configure do |c|
+            c.tracer.transport_options = ->(t) { t.adapter :unix, '/tmp/trace.sock' }
+          end
+        end
+
+        it { is_expected.to include agent_url: include('unix') }
+        it { is_expected.to include agent_url: include('/tmp/trace.sock') }
+      end
+
+      context 'with integrations loaded' do
+        before { Datadog.configure { |c| c.use :http, options } }
+        let(:options) { {} }
+
+        it { is_expected.to include integrations_loaded: start_with('http') }
+
+        it do
+          # Because net/http is default gem, we use the Ruby version as the library version.
+          is_expected.to include integrations_loaded: end_with("@#{RUBY_VERSION}")
+        end
+
+        context 'with integration-specific settings' do
+          let(:options) { { service_name: 'my-http' } }
+
+          it { is_expected.to include http_analytics_enabled: false }
```

Comment:
small nit, the suggested property name is `integration_<integration>_analytics_enabled` per the google doc, I believe. Not sure how strict we need to be toward the doc, feel free to disregard if it's not an issue.

Reply:
Nice catch, I think I was reading the wrong column in the spreadsheet when I named integration settings. Addressed!